crates/typst-library/src/compute/calc.rs

```diff
@@ -188,6 +189,34 @@ pub fn sqrt(
     Ok(value.v.float().sqrt())
 }
 
+/// Calculates the nth root of a number.
+///
+/// ```example
+/// #calc.root(4, 16) \
+/// #calc.root(3, 27)
+/// ```
+#[func]
+pub fn root(
+    /// Which root of the radicand to take
+    index: Spanned<i64>,
+    /// The expression to take the root of
+    radicand: Spanned<Num>,
+) -> SourceResult<f64> {
+    if index.v == 0 {
+        bail!(index.span, "cannot take the 0th root of a number");
+    } else if radicand.v.float() < 0. {
+        if index.v % 2 == 0 {
+            bail!(index.span, "cannot take the even root of a negative number because the result it is not real");
+        } else {
+            assert!(index.v % 2 == 1);
+            Ok(-(-radicand.v.float()).powf(1. / index.v as f64))
+        }
+    } else {
+        assert!(radicand.v.float() > 0.);
```

Comment:
what if `radicand.v.float() == 0.0`?

Reply:
I think that this assertion is also not necessary as e.g. `0.0_f32.powf(0.4))` works fine. If there is any doubt of what values the arguments could have, a `match` case could instead have been used to get a better structure.